### PR TITLE
Add x86_64 centos7 build with 'make centos7_in_docker' make target.

### PR DESCRIPTION
**Pull request by tomkraljevic**
Thank you for taking time to improve this repository!

### Before submitting a PR

Please make sure that:

- [ ] It is not a duplicate.
- [ ] There is an issue in the tracker for your PR. If there isn't one please make one first.
- [ ] Your changes are in a separate branch. Branches should have descriptive names (e.g. starting with `fix/` or `feature/` for example `fix/memory_allocation`).
- [ ] Commit messages adhere to our contributor guide.
- [ ] There is only a single commit (otherwise squash them).
- [ ] Project builds, all the tests are passing for you locally and the general rules from our [CONTRIBUTING.md](CONTRIBUTING.md) are met. Otherwise make changes and amend your commit (`git commit --amend`).

### Submitting a PR
  
Great, you are now ready to open a new PR!

- [ ] Come up with a descriptive, short, title.
- [ ] Write a short (but meaningful) summary explaining the changes.
- [ ] Be sure to add `closes #XXX` in the comment to auto-close the issue that the PR fixes.
- [ ] **Remove this template before submitting.**

### CODE REVIEW

**Comment by mdymczyk** [comment]
Ok I see only 1 thing to improve (the sklearn -> h2o4gpu rename) I commented on and rest looks good. Any other comments @pseudotensor ? You were mentioning some hardcoded paths?

File: Makefile [hunk 1]
```diff
@@ -657,5 +657,53 @@ ifeq ($(CI),)
...
+         export OPENBLAS_PREFIX="open" && \
+         scl enable devtoolset-3 "make fullinstalljenkins-nonccl-cuda8")
+	cp /tmp/build/src/interface_py/dist2/h2o4gpu*.whl dist
+	chmod o+rw dist/h2o4gpu*.whl
+
+centos7:
+	$(MAKE) centos7_setup
+	$(MAKE) centos7_build
+
+centos7_in_docker:
```
Comment (pseudotensor) [inline Makefile]: These makefile additions mean there is a separate path for centos compared to before.  If we decide centos is a way to go to get a base compatible version, everything should be redone like originally with scripts for different builds (nccl, cuda, etc.)

File: src/Makefile [hunk 2]
```diff
@@ -60,12 +60,12 @@ ifeq ($(USENVTX),1)
 DFLAGS+=-DUSE_NVTX=1
 endif
 
-CUDFLAGS=$(DFLAGS) -DHAVECUDA -DCUDCHECKERR -DCUDA_MAJOR=$(CUDA_MAJOR) -DCUDA_MINOR=$(CUDA_MINOR)
+CUDFLAGS=$(DFLAGS) -DHAVECUDA -DCUDCHECKERR -DCUDA_MAJOR=$(CUDA_MAJOR) -DCUDA_MINOR=$(CUDA_MINOR) -I/usr/include/openblas
```
Comment (pseudotensor) [inline src/Makefile]: Not generally correct, can't be merged.

File: tests/cpp/Makefile [hunk 3]
```diff
@@ -50,7 +50,7 @@ $(UTEST_OBJ_ROOT)/%.o: $(UTEST_ROOT)/%.cu
 	@mkdir -p $(@D)
 	$(CUXX) $(CUFLAGS) -I$(GTEST_INC) -o $@ -c $<
 
-UTEST_LIBS=-fopenmp -lcudart -lcudadevrt -lcublas -lcusolver -lcusparse -lblas -lcurand
+UTEST_LIBS=-fopenmp -lcudart -lcudadevrt -lcublas -lcusolver -lcusparse -l$(OPENBLAS_PREFIX)blas -lcurand
```
Comment (pseudotensor) [inline tests/cpp/Makefile]: can't be merged.

File: src/Makefile [hunk 4]
```diff
@@ -60,12 +60,12 @@ ifeq ($(USENVTX),1)
...
 ifeq ($(USENVTX),1)
 CUDFLAGS+=-I/usr/local/cuda/include
 endif
 
-CXXFLAGS=$(IFLAGS) $(OPTFLAG) $(HASH) -Wall -std=c++11 -fPIC $(LINKFLAGS) $(RINCLUDEHOME) $(DFLAGS)
+CXXFLAGS=$(IFLAGS) $(OPTFLAG) $(HASH) -Wall -std=c++11 -fPIC $(LINKFLAGS) $(RINCLUDEHOME) $(DFLAGS) -I/usr/include/openblas
```
Comment (pseudotensor) [inline src/Makefile]: Not generally correct, can't be merged.

File: examples/cpp/Makefile.gpu [hunk 5]
```diff
@@ -35,7 +35,7 @@ GCCSTUFF=
 endif
 
 ifeq ($(USEMKL),0)
-MKLSTUFF=-lm -lblas#using GPU instead for blas
+MKLSTUFF=-lm -l$(OPENBLAS_PREFIX)blas#using GPU instead for blas
```
Comment (pseudotensor) [inline examples/cpp/Makefile.gpu]: Where is this prefix set?  If not set, won't work in general.

**Pull request merged**